Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/ObjCStubs.kt

```diff
@@ -587,12 +587,13 @@ private class ObjCPropertyStubBuilder(
     }
 }
 
-fun ObjCClassOrProtocol.kotlinClassName(isMeta: Boolean): String {
+fun ObjCClassOrProtocol.kotlinClassName(isMeta: Boolean, isJ2ObjC: Boolean = false): String {
     val baseClassName = when (this) {
```

Comment:
For the code the way you've shaped it you'd want something like this:

Something like:
val baseClassName = when (this)
        is ObjCClass -> this.name
        is ObjCProtocol -> if (isJ2ObjC) this.name else "${this.name}Protocol"
}

Hover I'd add name override to ObjCProtol as opposed to a variable that needs to be passed around.

Reply:
Done. I looked into adding it to ObjCProtocol, but it didn't seem any more feasible. I would have to modify ObjCProtocolImpl, ObjCProtocol, and ObjCClassOrProtocol to pass down the information to this extension, so I don't think it's worth it?